chore(docs): fixup threads intro based on example (#166)
textileio/docs: docs/threads/index.md

ThreadDB is a p2p database built on [IPFS](https://ipfs.io) and [Libp2p](https://libp2p.io). Together, the Threads Protocol and Database provide an alternative architecture for data on the web. ThreadDB aims to help power a new generation of web technologies by combining a novel use of event sourcing, Interplanetary Linked Data ([IPLD](https://ipld.io)), and access control to provide a distributed, scalable, and flexible database solution for decentralized applications.

### How to use ThreadDB

There are two implementations of ThreadDB. 

#### Golang

> The first is written in Go and can be found at [https://github.com/textileio/go-threads/](https://github.com/textileio/go-threads/). This is the reference implementation, it contains all the latest components to use as a library, write trustless services, [build a client](https://godoc.org/github.com/textileio/go-threads/api/client) connected to a a threads daemon. More documentation for the Go implementation will be coming in the future.

Reviewer: ```suggestion
The first is written in Go and can be found at [https://github.com/textileio/go-threads/](https://github.com/textileio/go-threads/). This is the reference implementation, it contains all the latest components to use as a library, write trustless services, [build a client](https://godoc.org/github.com/textileio/go-threads/api/client) that connects to a threads daemon, etc. More documentation for the Go implementation will be coming in the future.
```

#### JavaScript

The second implementation is written in JavaScript (Typescript, really). This implementation has some optimizations to make it more ideal when writing web applications. The JavaScript implementation is best understood through the [Client](https://textileio.github.io/js-hub/docs/hub.client) and [Database](https://textileio.github.io/js-threads/modules/_textile_threads_database.html) documentation.

> In general, when you are building apps that use threads in remote context (e.g. the browser) it's best to push the networking later to remote services whenever possible (while using allowing p2p when it works). For this, we recommend using `js-threads` using the Hub's APIs. Using the [Database](https://textileio.github.io/js-threads/modules/_textile_threads_database.html) for example, you can perform all signing and encryption locally, keeping thread ownership with your users, but push persistence and networking to the remote Hub. You can also build your own remote relays and services using the [go-threads](https://github.com/textileio/go-threads/) library.

Reviewer: ```suggestion
In general, when you are building apps that use threads in remote context (e.g. the browser) it's best to push the networking later to remote services whenever possible (while using/allowing p2p when it works). For this, we recommend using `js-threads` using the Hub's APIs. Using the [Database](https://textileio.github.io/js-threads/modules/_textile_threads_database.html) for example, you can perform all signing and encryption locally, keeping thread ownership with your users, but push persistence and networking to the remote Hub. You can also build your own remote relays and services using the [go-threads](https://github.com/textileio/go-threads/) library.
```

For the rest of the explanation below, we'll focus on examples using the JavaScript library.

### Developer API

ThreadDB is designed to be simple enough for any developer to start using. The API will feel familiar to developers who have worked with technologies like MongoDB.

#### Data organization

> The first three concepts a developer will encounter with ThreadDB are [Databases](#databases), [Collections](#collections), and [Instances](#instances). The organization is simple. Instances are the individual records you create, update, or delete. Instances are stored in a Collection. Collections have one or many Schemas and can only store Instances that match one of those Schemas. Databases can store many Collections. Collections are similar to Tables in other databases. A Thread-based Database is tied to a single Thread (with associated Thread ID).

Reviewer: ```suggestion
The first three concepts a developer will encounter with ThreadDB are [Databases](#databases), [Collections](#collections), and [Instances](#instances). Instances are the individual records you create, update, or delete. Instances are stored in a Collection. Collections have one or many Schemas and can only store Instances that match one of those Schemas. Databases can store many Collections. Collections are similar to Tables in other databases. A Thread-based Database is tied to a single Thread (with associated Thread ID).
```

Reviewer: I just remembered from my paper-writing days that this is a bit of a no no (I know I had it in before). It alienates people for which this _isn't_ simple.

#### Creating a new database

> To start a new, empty database is simple. You can run the Database with remote networking using the Hub APIs. To do so, you initialize your database with the `UserAuth` object. You can read more about creating `UserAuth` objects in [the creating web apps tutorial](../tutorials/hub/web-app.md).

Reviewer: ```suggestion
To start a new, empty Database, with remote networking using the Hub APIs you simply initialize your database with the `UserAuth` object. You can read more about creating `UserAuth` objects in [the creating web apps tutorial](../tutorials/hub/web-app.md).
```

import { Database, UserAuth } from '@textile/threads'

async function create (auth: UserAuth, name: string) {
  const db = Database.withUserAuth(auth, name)
  return db
}
```

<!--
Depends on: https://github.com/textileio/js-threads/issues/293

The above assumes you are running the database in insecure environments such as the browser, but if you are running the database in a secure environment you can skip the tutorial and provide you [API keys](../hub/app-apis.md) directly.

```typescript
import { Database, KeyInfo } from '@textile/threads'

async function create (keys: KeyInfo, name: string) {
  // const db = Database.withKeyInfo(keys, name)
  // return db
}
```
-->

With your database created, you now need to *start* it. Starting a database requires the data owner's identity. In these examples, we'll use a [random public-key infrastructure (PKI) based identity](../tutorials/hub/libp2p-identities.md), however, see [Identity](#identity) for further details and links.

```typescript
import { Identity } from '@textile/threads-core'
import { Database, UserAuth, ThreadID } from '@textile/threads'

async function start (db: Database, identity: Identity) {

>   const randomID = ThreadID.fromRandom()
>   await db.start(identity, {threadID: randomID})

Reviewer: ```suggestion
  const threadID = ThreadID.fromRandom()
  await db.start(identity, { threadID })
```
